loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ShuntCompensatorsValidation.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2017, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.loadflow.validation;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.ShuntCompensator;
+import com.powsybl.loadflow.validation.io.ValidationWriter;
+
+/**
+ *
+ * @author Massimo Ferraro <massimo.ferraro@techrain.eu>
+ */
+public final class ShuntCompensatorsValidation {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StaticVarCompensatorsValidation.class);
+
+    private ShuntCompensatorsValidation() {
+    }
+
+    public static boolean checkShunts(Network network, ValidationConfig config, Path file) throws IOException {
+        Objects.requireNonNull(file);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(file);
+        try (Writer writer = Files.newBufferedWriter(file, StandardCharsets.UTF_8)) {
+            return checkShunts(network, config, writer);
+        }
+    }
+
+    public static boolean checkShunts(Network network, ValidationConfig config, Writer writer) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(writer);
+        try (ValidationWriter shuntsWriter = ValidationUtils.createValidationWriter(network.getId(), config, writer, ValidationType.SHUNTS)) {
+            return checkShunts(network, config, shuntsWriter);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static boolean checkShunts(Network network, ValidationConfig config, ValidationWriter shuntsWriter) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(shuntsWriter);
+        LOGGER.info("Checking shunt compensators of network {}", network.getId());
+        return network.getShuntStream()
+                      .sorted(Comparator.comparing(ShuntCompensator::getId))
+                      .map(shunt -> checkShunts(shunt, config, shuntsWriter))
+                      .reduce(Boolean::logicalAnd)
+                      .orElse(true);
+    }
+
```

Comment:
These 3 methods are duplicated in all the *Validation classes. I think we have to do a refactor to avoid the duplication. It should be done in a different PR